src/core/socialprovider.js

```diff
@@ -122,6 +122,14 @@ XMPPSocialProvider.prototype.initializeState = function() {
  * @param {Function} continuation Callback upon connection
  */
 XMPPSocialProvider.prototype.connect = function(continuation) {
+  if (this.client) {
```

Comment:
Do we have a sense of when this is triggered? My intuition is that if connect() is called when this.client exists, that's representative of a bug.